sdk/iot/core/src/az_iot_telemetry.c

```diff
@@ -27,8 +27,8 @@ AZ_NODISCARD az_result az_iot_hub_client_telemetry_publish_topic_get(
   AZ_PRECONDITION_VALID_SPAN(mqtt_topic, 0, false);
   AZ_PRECONDITION_NOT_NULL(out_mqtt_topic);
 
-  const az_span* user_agent = &(client->_internal.options.user_agent);
-  const az_span* module_id = &(client->_internal.options.module_id);
+  const az_span* const user_agent = &(client->_internal.options.user_agent);
+  const az_span* const module_id = &(client->_internal.options.module_id);
```

Comment:
This is probably a naive question, but how did you know where to apply the `const` to?

Reply:
No questions are too naive :)

Here is essentially how it breaks down. I'll use char* as an example:
```C
char* const x
```
is a constant pointer to a char. So you can change the char at `*x` but not the value of `x`.

```C
const char* x
```
is a pointer to a constant char. So you can change the pointer but not the content at `*x`.

```C
const char* const x
```
is a constant pointer to a constant char. So you can't change the value of the pointer nor the content at `*x`.

[This website](https://cdecl.org) is really great to understand pretty much any declaration in C. I included links to my examples.

https://cdecl.org/?q=const+char*+const+x
https://cdecl.org/?q=char*+const+x
https://cdecl.org/?q=const+char*+x